instrumentation/vertx/vertx-rx-java-3.5/javaagent/src/version5Test/java/io/opentelemetry/javaagent/instrumentation/vertx/reactive/server/VertxReactivePropagationTest.java

```diff
@@ -0,0 +1,229 @@
+
+class VertxReactivePropagationTest {
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @RegisterExtension static final AutoCleanupExtension cleanup = AutoCleanupExtension.create();
+
+  private static WebClient client;
+  private static int port;
+  private static Vertx server;
+
+  @BeforeAll
+  static void setUp() throws ExecutionException, InterruptedException, TimeoutException {
+    port = PortUtils.findOpenPort();
+    server = VertxReactiveWebServer.start(port);
+    client = WebClient.of("h1c://localhost:" + port);
+  }
+
+  @AfterAll
+  static void cleanUp() {
+    server.close();
+  }
+
+  // Verifies that context is correctly propagated and sql query span has correct parent.
+  // Tests io.opentelemetry.javaagent.instrumentation.vertx.reactive.VertxRxInstrumentation
+  @SuppressWarnings("deprecation") // uses deprecated db semconv
+  @Test
+  void contextPropagation() {
+    AggregatedHttpResponse response = client.get("/listProducts").aggregate().join();
+    assertThat(response.status().code()).isEqualTo(SUCCESS.getStatus());
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("GET /listProducts")
+                        .hasKind(SpanKind.SERVER)
+                        .hasNoParent()
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(NETWORK_PROTOCOL_VERSION, "1.1"),
+                            equalTo(NETWORK_PEER_ADDRESS, "127.0.0.1"),
+                            satisfies(NETWORK_PEER_PORT, val -> val.isInstanceOf(Long.class)),
+                            equalTo(SERVER_ADDRESS, "localhost"),
+                            equalTo(SERVER_PORT, port),
+                            equalTo(CLIENT_ADDRESS, "127.0.0.1"),
+                            equalTo(URL_PATH, "/listProducts"),
+                            equalTo(HTTP_REQUEST_METHOD, "GET"),
+                            equalTo(HTTP_RESPONSE_STATUS_CODE, 200),
+                            equalTo(URL_SCHEME, "http"),
+                            satisfies(USER_AGENT_ORIGINAL, val -> val.isInstanceOf(String.class)),
+                            equalTo(HTTP_ROUTE, "/listProducts")),
+                span ->
+                    span.hasName("handleListProducts")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.hasName("listProducts")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(1)),
+                span ->
+                    span.hasName("SELECT products")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasParent(trace.getSpan(2))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(
```

Comment:
Unlike the same test in older versions this span comes from `vertx-sql-client` instrumentation. The way it is set up in this test it is not able to capture a bunch of attributes like `db.user`, `db.system`, `db.name` etc.